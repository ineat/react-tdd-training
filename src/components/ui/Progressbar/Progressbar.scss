#progressbar {
  width: 300px;
  margin: auto;
  margin-top: 1rem;
  margin-bottom: 2rem;
	overflow: hidden;
  counter-reset: step;

  li {
    list-style-type: none;
    color: lightslategrey;
    text-transform: uppercase;
    font-size: 9px;
    width: 33.33%;
    float: left;
    position: relative;

    &:before {
      content: counter(step);
      counter-increment: step;
      width: 20px;
      line-height: 20px;
      display: block;
      font-size: 10px;
      color: white;
      background: lightslategrey;
      border-radius: 3px;
      margin: 0 auto 5px auto;
    }

    &:after {
      content: '';
      width: 100%;
      height: 2px;
      background: lightslategrey;
      position: absolute;
      left: -50%;
      top: 9px;
      z-index: -1;
    }

    &:first-child:after {
      content: none;
    }
  }

  li.active {
    &:after, &:before {
      background: lightgreen;
      color: white;
    }
  }
}