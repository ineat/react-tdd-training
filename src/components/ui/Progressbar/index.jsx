import * as React from 'react';

import './Progressbar.scss';

export default ({ activeStepIndex }) => {
  const getStepIndicatorClass = (index, activeIndex) =>
    index === activeIndex ? 'step-indicator active' : 'step-indicator';

  const steps = [{
    index: 1,
    label: 'Animal'
  }, {
    index: 2,
    label: 'Breed'
  }, {
    index: 3,
    label: 'Picture'
  }];

  return (
    <ul id="progressbar">
      {
        steps.map(step =>
          <li className={getStepIndicatorClass(step.index, activeStepIndex)} key={step.index}>
            {step.label}
          </li>
        )
      }
    </ul>
  );
};