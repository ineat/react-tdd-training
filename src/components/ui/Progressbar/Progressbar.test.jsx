import * as React from 'react';
import { mount, shallow } from 'enzyme';

import Progressbar from '.';

describe('Progressbar', () => {
  it('Should render.', () => {
    expect(mount(<Progressbar activeStepIndex={1} />)).toMatchSnapshot();
  });

  it('Should highlight the first step when the given active step index is 1.', () => {
    const wrapper = shallow(<Progressbar activeStepIndex={1} />);

    const steps = wrapper.find('.step-indicator');

    expect(steps.at(0).hasClass('active')).toEqual(true);
    expect(steps.at(1).hasClass('active')).toEqual(false);
    expect(steps.at(2).hasClass('active')).toEqual(false);
  });

  it('Should highlight the second step when the given active step index is 2.', () => {
    const wrapper = shallow(<Progressbar activeStepIndex={2}/>);

    const steps = wrapper.find('.step-indicator');

    expect(steps.at(0).hasClass('active')).toEqual(false);
    expect(steps.at(1).hasClass('active')).toEqual(true);
    expect(steps.at(2).hasClass('active')).toEqual(false);
  });

  it('Should highlight the third step when the given active step index is 3.', () => {
    const wrapper = shallow(<Progressbar activeStepIndex={3}/>);

    const steps = wrapper.find('.step-indicator');

    expect(steps.at(0).hasClass('active')).toEqual(false);
    expect(steps.at(1).hasClass('active')).toEqual(false);
    expect(steps.at(2).hasClass('active')).toEqual(true);
  });
});