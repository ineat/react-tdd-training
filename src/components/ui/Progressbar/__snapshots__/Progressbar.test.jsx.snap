// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progressbar Should render. 1`] = `
<_default
  activeStepIndex={1}
>
  <ul
    id="progressbar"
  >
    <li
      className="step-indicator active"
      key="1"
    >
      Animal
    </li>
    <li
      className="step-indicator"
      key="2"
    >
      Breed
    </li>
    <li
      className="step-indicator"
      key="3"
    >
      Picture
    </li>
  </ul>
</_default>
`;
